src/lib/types.h

```diff
@@ -0,0 +1,10 @@
+#ifndef LIB_TYPES_H
```

Comment:
this is a new file now and would need to be covered by Makefile.am. I think what you previously did was great. Just include stdint.h and have the typedefs adjusted.